File: api/src/main/java/org/pmiops/workbench/config/CommonConfig.java
package org.pmiops.workbench.config;

import static com.google.api.client.googleapis.util.Utils.getDefaultJsonFactory;

import com.google.api.client.json.JsonFactory;
import com.google.common.base.Stopwatch;
import java.security.SecureRandom;
import java.time.Clock;
import java.util.Random;
import org.javers.core.Javers;
import org.javers.core.JaversBuilder;
import org.springframework.beans.factory.config.ConfigurableBeanFactory;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Scope;

@Configuration
public class CommonConfig {
  @Bean
  JsonFactory jsonFactory() {
    return getDefaultJsonFactory();
  }

  @Bean
  Clock clock() {
    return Clock.systemUTC();
  }

  @Bean
  Random random() {
    return new SecureRandom();
  }

  @Bean
  Javers javers() {
    return JaversBuilder.javers().build();
  }

  @Bean
  @Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
  public Stopwatch getStopwatch() {
    return Stopwatch.createUnstarted();
  }
}

File: api/src/main/java/org/pmiops/workbench/config/CommonConfig.kt
package org.pmiops.workbench.config

import com.google.api.client.googleapis.auth.oauth2.GoogleCredential
import com.google.api.client.json.JsonFactory
import com.google.api.client.json.jackson2.JacksonFactory
import java.security.SecureRandom
import java.time.Clock
import java.util.Random
import org.springframework.beans.factory.config.ConfigurableBeanFactory
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Scope

@Configuration
class CommonConfig {

    @Bean
    internal fun jsonFactory(): JsonFactory {
        return JacksonFactory()
    }

    @Bean
    internal fun clock(): Clock {
        return Clock.systemUTC()
    }

    @Bean
    internal fun random(): Random {
        return SecureRandom()
    }

    @Bean
    @Scope(value = ConfigurableBeanFactory.SCOPE_PROTOTYPE)
    internal fun googleCredentialBuilder(): GoogleCredential.Builder {
        return GoogleCredential.Builder()
    }
}
